src/pds/ingress/service/pds_status_app.py

```diff
@@ -0,0 +1,295 @@
+    file_info : dict
+        Dictionary containing file metadata dervied from the requestor's version
+        of the file. This includes file size, an md5 hash, and the last modified time.
+
+    Returns
+    -------
+    ingress_status : str
+        One of the ingress status string values described above.
+
+    """
+    try:
+        object_head = s3_client.head_object(Bucket=destination_bucket, Key=object_key)
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "404":
+            # File does not exist in S3
+            ingress_status = "Missing"
+            return ingress_status
+        else:
+            # Some other kind of unexpected error
+            raise
+
+    object_length = int(object_head["ContentLength"])
+    object_last_modified = object_head["Metadata"]["last_modified"]
+    object_md5 = object_head["Metadata"]["md5"]
+
+    request_length = file_info["size"]
+    request_last_modified = file_info["last_modified"]
+    request_md5 = file_info["md5"]
+
+    if not (
+        object_length == request_length and object_md5 == request_md5 and object_last_modified == request_last_modified
+    ):
+        ingress_status = "Modified"
+    else:
+        ingress_status = "Uploaded"
+
+    return ingress_status
+
+
+def send_email(message_body, return_email):
+    """
+    Sends the provided message body to the provided return email address.
+
+    Email is sent via AWS SMTP endpoint, which requires credentials pulled by
+    this function from the AWS SSM Parameter store.
+
+    Parameters
+    ----------
+    message_body : str
+        Body of the email message to send.
+    return_email : str
+        Email address to send to.
+
+    Raises
+    ------
+    RuntimeError
+        If the expected SMTP endpoint parameters are not successfully pulled
+        from the SSM Parameter store.
+
+    """
+    smtp_config = {}
+
+    # Pull the required SMTP endpoint parameters from SSM
+    smtp_config_ssm_key_path = os.environ["SMTP_CONFIG_SSM_KEY_PATH"]
+    response = ssm_client.get_parameters_by_path(Path=smtp_config_ssm_key_path, Recursive=True)
+
+    # Map SSM parameter values to unique portion of the SSM key name
+    for ssm_parameter in response["Parameters"]:
+        smtp_config[ssm_parameter["Name"].split("/")[-1]] = ssm_parameter["Value"]
+
+    expected_fields = ("username", "password", "server", "sender")
+
+    if not all(field in smtp_config for field in expected_fields):
+        raise RuntimeError(
+            f"Unexpected SMTP configuration from SSM, " f"expected {expected_fields}, got {list(smtp_config.keys())}"
+        )
+
+    # Create the email payload
+    message = MIMEText(message_body)
+    message["Subject"] = "PDS Data Upload Manager Status Service Request"
+    message["From"] = smtp_config["sender"]
+    message["To"] = return_email
+
+    # Send the email via SMTP endpoint
+    endpoint_host, endpoint_port = smtp_config["server"].split(":")
```

Comment:
Will the "server" entry ever just be a bare hostname (default port), like `smtp.jpl.nasa.gov`, no `:`?

Reply:
Since the server and port are dictated to us from Cloud Computing Services (CCS), I think its safe to assume it will always be there.